TestClients/NICE.Identity.TestClient.NETCore/Program.cs

```diff
@@ -1,17 +1,36 @@
-﻿using Microsoft.AspNetCore;
+﻿using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 
 namespace NICE.Identity.TestClient.NETCore
 {
 	public class Program
 	{
 		public static void Main(string[] args)
 		{
-			CreateWebHostBuilder(args).Build().Run();
-		}
+			Func<IHostingEnvironment, IConfigurationBuilder> configurationFactory = env =>
+				new ConfigurationBuilder()
+					.SetBasePath(env.ContentRootPath)
+					.AddEnvironmentVariables()
+					.AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+					//.AddJsonFile($"appsettings.{environmentName}.json", optional: true)
+					.AddUserSecrets(Assembly.GetAssembly(typeof(Startup)));
+
+            var startup = new Startup("Publications", configurationFactory, ProductionDependencies.AddProductionDependencies);
```

Comment:
i'm not sure about this line. why is this needed?

Reply:
You need this line to add production dependencies but also swap out the dependencies for mocked objects for testing.